### ISSUE TITLE
Page zoom is restricted - uids.brand.uiowa.edu

### ISSUE THREAD

**Issue by briand44** [uiowa/uids#1039]
# Motivation

Siteimprove is reporting "Page zoom is restricted". This is a WCAG 2.1 AA compliance issue.

Note from Marie - Page zoom is restricted - most pages, specifically regarding the viewport. Note: Zooming in seems to work fine for me.

# Siteimprove info

**Technical doc**
https://alfa.siteimprove.com/rules/sia-r47

Siteimprove link to existing issue
- https://my2.siteimprove.com/Inspector/997947/A11y/Page?pageId=24734804921&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r47/failed/eUV3bjRON2U=;ZjA1ODMxZGY=
- https://my2.siteimprove.com/Inspector/997947/A11y/Page?pageId=69552757820&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r47/failed/eUV3bjRON2U=;ZjA1ODMxZGY=
- https://my2.siteimprove.com/Inspector/997947/A11y/Page?pageId=69552798044&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r47/failed/eUV3bjRON2U=;ZjA1ODMxZGY=

Example
- https://uids.brand.uiowa.edu
- https://uids.brand.uiowa.edu/?path=/docs/introduction--docs
- https://uids.brand.uiowa.edu/?path=/story/not-implemented-slider--slider

**Page zoom is restricted**
Issue
1.4.10: Reflow
1.4.4: Resize text
The viewport element on this page prevents users from zooming in or scaling text content.

**Comment by briand44** [comment]
Looks like it is related to `<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">`

Specifically: `maximum-scale=1`

> For each test target, the content attribute does not:
> - specify the property user-scalable with a value of fixed; nor
> - specify the property maximum-scale with a value of less than 2.

**Comment by pyrello** [comment]
This seems to be part of a generated file. The only instance I find of `maximum-scale=1` is in the generated `index.html` file.

**Comment by briand44** [comment]
## Grooming
 
**Quadrant:** 

Complex - unsure of how/where this is being added and unsure of how to fix

 **Discussion:**
 - Marie was able to Zoom in her testing but Siteimprove still throwing the issue
   - https://www.w3.org/WAI/standards-guidelines/act/rules/b4f0c3/ - may only affect old mobile browsers
 - Seems related to `maximum-scale=1` https://github.com/uiowa/uids/issues/1039#issuecomment-4157266408
 - Investigate to get some understanding of what the viewport `maximum-scale` does and best practices around this
 - Unsure of where/how this attribute is getting added
 - https://github.com/storybookjs/storybook/issues/31272 - do we need a storybook update?

**Proposed solution or definition of done:** 
- See if storybook update resolves issue, if not...
- Investigate to understand where the `maximum-scale=1` is being added to the viewport meta tag
- Implement fix so that Siteimprove no longer reports this as an issue

**Comment by joewhitsitt** [comment]
>See if storybook update resolves issue, if not...

So to clarify, we want to try the update first here? See if the update is worth the squeeze and may fix the other issue that were planned?

Is there any concurrent work in/surrounding uids that would be affected by an upgrade PR?

**Comment by joewhitsitt** [comment]
Just looking at numbers I think storybook 9+ where this is fixed requires node 20+ which caused us to pause https://github.com/uiowa/uiowa/issues/9534 no?

>I would love some thoughts from the team here, it seems like the styles we are using form the old version of uids are not compatible with node 22, so we need to import the scss with no scripts. Is that okay? do we need to update UIDS 4.x with the relevant things and re-organize?

>We would need to update the viewbook scss completely to ensure this is fully done.
>https://github.com/uiowa/admissions-viewbook/pull/83

**Comment by joewhitsitt** [comment]
Maybe I am reading this wrong because the viewbooks consume the 3.x version of UIDS so it shouldn't be affected by a 4.x branch change?

**Comment by joewhitsitt** [comment]
Going to see how far I get

**Comment by joewhitsitt** [comment]
So, i've gotten far enough with an upgrade locally to know that the stable 9.1.20 version solves this zoom issue with the siteimprove widget. Digesting and working through some upgrade pains though https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-8x-to-900 to get everything fully working.

**Comment by pyrello** [comment]
> Maybe I am reading this wrong because the viewbooks consume the 3.x version of UIDS so it shouldn't be affected by a 4.x branch change?

This scared me for a few moments because we removed the `3.x` branch. But the viewbooks point at specific 3.x tagged releases. Not ideal, but at least not broken by definition.

**Comment by joewhitsitt** [comment]
`main` is essentially `3.x` too. again, not ideal

**Comment by joewhitsitt** [comment]
@briand44 found another A issue that isn't auto fixed by the upgrade. All the select controls do not have a aria-labelledby type relationship to the label in the next door table cell (e.g. Type)

<img width="1177" height="788" alt="Image" src="https://github.com/user-attachments/assets/05f166aa-54c5-4408-8c6b-c3fbda99c589" />
